peakinvestigator/progress/uploader.py

```diff
@@ -49,56 +48,45 @@ def upload_filehandle(self, num, filetype, size, handle, progress=None):
     def upload_file(self, filename, num, with_progress=True):
         size = os.path.getsize(filename)
         filetype = 0 if filename[-4:] == '.bin' else 1
-        if self.progress_factory == None:
-            with io.open(filename, 'rb') as f:
-                self.upload_filehandle(num, filetype, size, f)
-        else:
-            progress = self.progress_factory.create(total=size, unit='bytes') if with_progress else None
-            with io.open(filename, 'rb') as f:
-                self.upload_filehandle(num, filetype, size, f, progress)
+        progress = self.progress_factory.create(total=size, unit='bytes') if (self.progress_factory != None and with_progress) else None
+        with io.open(filename, 'rb') as f:
+            self.upload_filehandle(num, filetype, size, f, progress)
+        if (with_progress and progress):
```

Comment:
Seems like `with_progress` is redundant. From a few lines above, `progress` will be None if `with_progress` is `False`.

Reply:
This is corrected on latest commit.